sources/packages/web/src/services/InstitutionService.ts

```diff
@@ -0,0 +1,67 @@
+import { Institute } from "../types/contracts/InstituteContract";
+import { EducationProgram, InstitutionProfileState } from "../types";
+import ApiClient from "./http/ApiClient";
+
+export class InstitutionService {
+  // Share Instance
+  private static instance: InstitutionService;
+
+  public static get shared(): InstitutionService {
+    return this.instance || (this.instance = new this());
+  }
+
+  public async getInstitutes(): Promise<Institute[]> {
+    // TODO: Sending dummy list for now later replace with API call
+    const institutes: Institute[] = [
+      {
+        name: "Sprott Shaw College",
+        code: "ssc",
+      },
+    ];
+    return institutes;
+  }
+
+  public async getProgramsFor(
+    institute?: Institute,
+  ): Promise<EducationProgram[]> {
+    if (institute?.code === "ssc") {
+      const programs: EducationProgram[] = [
+        {
+          name: "Nursing Program",
+          code: "np",
+        },
+      ];
+      return programs;
+    }
+    return [];
+  }
+
+  public async createInstitution(institutionProfile: InstitutionProfileState) {
+    const institutionDto = {
```

Comment:
Is there any specific reason for no use of type?

Reply:
Changed!